document/src/vespa/document/update/addfieldpathupdate.cpp

```diff
@@ -53,13 +54,13 @@ class AddIteratorHandler : public fieldvalue::IteratorHandler {
 
 ModificationStatus
 AddIteratorHandler::doModify(FieldValue &fv) {
-    if (fv.inherits(CollectionFieldValue::classId)) {
+    if (fv.isCollection()) {
         CollectionFieldValue &cf = static_cast<CollectionFieldValue &>(fv);
         for (std::size_t i = 0; i < _values.size(); ++i) {
             cf.add(_values[i]);
         }
     } else {
-        vespalib::string err = make_string("Unable to add a value to a \"%s\" field value.", fv.getClass().name());
+        vespalib::string err = make_string("Unable to add a value to a \"%s\" field value.", vespalib::getClassName(fv).c_str());
```

Comment:
It feels slightly naughty to me to invoke RTTI and ABI-specific demangling to get to the class names of a small set of concrete types. Should we instead have a utility function on `FieldValue` to get the class/type name, either virtual or just a big ol' `switch` block based on `FieldValue::_type`?

Reply:
First option chosen with a switch block.